resources/lib/api.py

```diff
@@ -32,8 +32,19 @@ def addon_data_received(self, data, encoding='base64'):
         self.encoding = encoding
 
     @staticmethod
-    def play_kodi_item(episode):
-        jsonrpc(method='Player.Open', id=0, params=dict(item=dict(episodeid=episode.get('episodeid'))))
+    def play_kodi_item(episode, current_episodeid):
+        ''' Play the next local episode '''
+        # Reset resume_time for next episode so we force starting from the beginning
+        # jsonrpc(dict(method='VideoLibrary.SetEpisodeDetails', params=dict(episodeid=episode.get('episodeid'), resume=dict(position=0))))
+
+        # Play the next episode
+        jsonrpc(dict(method='Player.Open', params=dict(item=dict(episodeid=episode.get('episodeid')))))
+
+        # If we do this too quickly, it won't be effective :-(
+        sleep(200)
+
+        # Mark the current episode as watched and reset position
+        jsonrpc(dict(method='VideoLibrary.SetEpisodeDetails', params=dict(episodeid=current_episodeid, playcount=1, resume=dict(position=0))))
```

Comment:
We probably want to increase the existing playcount, rather than set it to 1 to indicate watched.

Reply:
Not to sure, from memory I think Kodi only cares about it being either 0 or 1 and I think some addons possibly check for it actually being 1 to mark as watched.

Reply:
Correct if you only consider the Watched flag. But I just recently made a pull request to add this information to the Video info pane so it is exposed to the user in Kodi Matrix :roll_eyes: 

So we might as well do the right thing here. Especially since we already have the playcount IIRC.

Reply:
Cool so my memory was correct! Yeah let’s go for it then 😉